components/brave_wallet_ui/page/screens/onboarding/components/recovery-phrase/recovery-phrase.tsx

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+
+// utils
+import { getLocale } from '$web-common/locale'
+import { unbiasedRandom } from '../../../../../utils/random-utils'
+
+// styles
+import {
+  RecoveryPhraseContainer,
+  RecoveryBubble,
+  FrostedGlass,
+  HiddenPhraseContainer,
+  EyeOffIcon,
+  RecoveryBubbleBadge
+} from './recovery-phrase.style'
+
+export interface SelectedPhraseWord {
+  index: number
+  value: string
+}
+
+interface Props {
+  hidden: boolean
+  recoveryPhrase: string[]
+  onClickReveal?: () => void
+  verificationModeEnabled?: boolean
+  onVerifyUpdate?: (doesWordOrderMatch: boolean) => void
+  onSelectedWordListChange?: (words: SelectedPhraseWord[]) => void
+}
+
+const FAKE_PHRASE_WORDS = new Array(12).fill('Fake')
+
+const SELECTED_WORD_ORDINALS = {
+  0: getLocale('braveWalletOrdinalFirst'),
+  1: getLocale('braveWalletOrdinalThird'),
+  2: getLocale('braveWalletOrdinalLast')
+}
+
+const getNewSelectedWordsList = (prevWords: SelectedPhraseWord[], word: SelectedPhraseWord): SelectedPhraseWord[] => {
+  // remove from list if word already selected
+  if (prevWords.find((prevWord) => prevWord.index === word.index)) {
+    return prevWords.filter((prevWord) => prevWord.index !== word.index)
+  }
+
+  // 3 words selected max
+  if (prevWords.length > 2) {
+    return prevWords
+  }
+
+  // add word to list
+  return [...prevWords, word]
+}
+
+export const RecoveryPhrase: React.FC<Props> = ({
+  hidden,
+  onClickReveal,
+  onVerifyUpdate,
+  onSelectedWordListChange,
+  recoveryPhrase,
+  verificationModeEnabled
+}) => {
+  // state
+  const [selectedWords, setSelectedWords] = React.useState<SelectedPhraseWord[]>([])
+
+  // methods
+  const makeOnClickWord = React.useCallback((word: SelectedPhraseWord) => () => {
+    if (verificationModeEnabled) {
+      setSelectedWords(prev => getNewSelectedWordsList(prev, word))
+    }
+  }, [verificationModeEnabled])
+
+  // memos
+  const shuffledPhrase: string[] = React.useMemo(() => {
+    const array = recoveryPhrase.slice().sort()
+
+    for (let i = array.length - 1; i > 0; i--) {
+      let j = unbiasedRandom(0, array.length - 1)
```

Comment:
TODO @kdenhartog: Figure out what's happening with unbiased random and why it's being used vs a standard random function